docs/designs/xref.md

```diff
@@ -0,0 +1,103 @@
+# Xref
+- User can define uid with some properties in markdown, JSON and YAML files. Then output xrefmap.json for others to reference to. 
+- User can reference to uids in markdown, JSON and YAML files, they will be resolved during `docfx build`.
+
+## Define uid
+- User can define uid in YAML header of markdown file
+```
+---
+uid: a
+title: ASP.NET Documentation
+description: Learn how to develop ASP.NET and ASP.NET web applications. Get documentation, example code, tutorials, and more.
+---
+```
+- User can also define uid in JSON
+```json
+{
+  "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+  "uid": "a",
+  "title": "ASP.NET Documentation",
+  "description": "Learn how to develop ASP.NET and ASP.NET web applications. Get documentation, example code, tutorials, and more."
+}
+```
+- And define uid in YAML
+```yml
+#YamlMime:TestData
+uid: a
+title: ASP.NET Documentation
+description: Learn how to develop ASP.NET and ASP.NET web applications. Get documentation, example code, tutorials, and more.
+```
+
+## Output xref map
+Docfx will output a JSON file named `xrefmap.json`. In V2, docfx used to output `xrefmap.yml`, it took much longer to be de-serialized compared to JSON format.
+```json
```

Comment:
xrefmap's schema seems shared between v2 and v3. Can we support json/yaml as input and json as output to provide better v3 compatibility?

Reply:
We are outputting xrefmap.json for xrefmap.yml in OPS for the compatibility

---
In reply to: [230991643](https://github.com/dotnet/docfx/pull/3578#discussion_r230991643) [](ancestors = 230991643)